docs/types.rst

```diff
@@ -413,8 +413,18 @@ function type should not return anything, the whole ``returns (<return types>)``
 part has to be omitted.
 
 By default, function types are internal, so the ``internal`` keyword can be
-omitted. In contrast, contract functions themselves are public by default,
-only when used as the name of a type, the default is internal.
+omitted. In contrast, visibility has to be specified for contract functions themselves,
+only when used as a type, there is a default of internal.
+
+A function type ``A`` is implicitly convertible to a function type ``B`` if and only if
+their parameter types are identical, their return types are identical,
+their internal/external property is identical and the state mutability of ``A``
```

Comment:
Maybe ``visibility`` instead of ``internal/external property``?

Reply:
Hm, this is not really about visibility. For function definitions, `internal/external` can be derived from the visibility, but for function types this is more specific and "visibilty" would be the wrong term.

Reply:
But yeah, we should probably find a term for this at some point...

Reply:
True.